activerecord/lib/active_record/autosave_association.rb

```diff
@@ -527,7 +527,7 @@ def compute_primary_key(reflection, record)
         elsif reflection.options[:query_constraints] && (query_constraints = record.class.query_constraints_list)
           query_constraints
         else
-          :id
+          record.class.primary_key
```

Comment:
I just wanted to add a bit of context on this change. This is something we have already discussed in 
https://github.com/rails/rails/pull/47378/files#r1104639481
and 
https://github.com/rails/rails/pull/47230#discussion_r1094665781

Hardcoded `:id` value was incorrect but was working because `:id` through `object.public_send(:id)` translates to whichever column being used as a primary key. So if the primary key is `title` column - `object.public_send(:id)` returns the title value. But it wasn't working well with a composite primary key where `id` column is a part of the key so in order for it to work properly we changed it to be dynamically derived so if the record's primary key is `title` the `compute_primary_key` method returns `"title"`